intervention_graph_creation/src/local_graph_extraction/extract.py

```diff
@@ -26,65 +48,82 @@ def __init__(self) -> None:
-    ) -> Tuple[Dict[str, Any], Dict[str, Any]]:
-        # https://platform.openai.com/docs/guides/structured-outputs
+    def call_openai(self, messages: List[Dict[str, Any]]) -> Any:
         response = self.client.responses.parse(
             model=MODEL,
             input=messages,
-            reasoning={
-                "effort": REASONING_EFFORT,
-                # "summary": "auto",
-            }
+            reasoning={"effort": REASONING_EFFORT}
         )
         return response
 
     def process_dir(self, input_dir: Path, first_n: Optional[int] = None) -> None:
         self.ensure_dir(OUTPUT_DIR)
-
         pdf_paths = sorted(input_dir.absolute().glob("*.pdf"))
         if first_n:
             pdf_paths = pdf_paths[:first_n]
-        processed_papers = set([x.stem for x in OUTPUT_DIR.glob("*.json")])
         for pdf_path in tqdm(pdf_paths):
-            if pdf_path.stem in processed_papers:
+            out_dir = OUTPUT_DIR / pdf_path.stem
+            if out_dir.exists():
                 continue
+            out_dir.mkdir(parents=True, exist_ok=True)
+            raw_path = out_dir / f"{pdf_path.stem}_raw_response.txt"
+            json_path = out_dir / f"{pdf_path.stem}.json"
+            summary_path = out_dir / f"{pdf_path.stem}_summary.txt"
+            try:
+                file_id = self.upload_pdf_get_id(pdf_path)
+                messages = self.build_user_input(file_id)
+                response = self.call_openai(messages)
+                try:
+                    raw_response = response.model_dump_json() if hasattr(response, "model_dump_json") else str(response)
+                except Exception:
+                    raw_response = str(response)
+                safe_write(raw_path, raw_response)
+                try:
+                    output_text = getattr(response, "output_text", "")
+                    if not isinstance(output_text, str):
+                        output_text = str(output_text or "")
+                except Exception:
+                    output_text = ""
+                text_part, json_part = split_text_and_json(output_text or "")
+                safe_write(summary_path, text_part or "")
+                if json_part:
+                    try:
+                        parsed = json.loads(json_part)
+                        safe_write(json_path, json.dumps(parsed, ensure_ascii=False, indent=2))
+                    except Exception as e:
+                        err = {"error": f"Invalid JSON extracted: {str(e)}"}
+                        safe_write(json_path, json.dumps(err, ensure_ascii=False, indent=2))
+                else:
+                    err = {"error": "No JSON block found in output_text"}
+                    safe_write(json_path, json.dumps(err, ensure_ascii=False, indent=2))
+            except Exception as e:
+                err_text = f"Processing failed for {pdf_path.name}\n{str(e)}\n\nTraceback:\n{traceback.format_exc()}"
```

Comment:
Overall too many try/except, we can move it all into parse.py or similar

Reply:
[I have removed.](https://github.com/MartinLeitgab/AISafetyIntervention_LiteratureExtraction/pull/51/commits/45606f58b4c8ad04aa54816fa0c526b7454f2848)